src/finch/symbolic/rewriters.py

```diff
@@ -58,24 +76,25 @@ class PreWalk:
     def __init__(self, rw: RwCallable):
         self.rw = rw
 
-    def __call__(self, x: Term) -> Term | None:
+    def __call__(self, x: T) -> T | None:
+        from ..finch_logic import LogicExpression
+
         y = self.rw(x)
-        if y is not None:
-            if y.is_expr():
-                args = y.children()
-                return y.make_term(
-                    y.head(), *[default_rewrite(self(arg), arg) for arg in args]
-                )
-            return y
-        if x.is_expr():
-            args = x.children()
-            new_args = list(map(self, args))
-            if not all(arg is None for arg in new_args):
-                return x.make_term(
-                    x.head(),
-                    *map(lambda x1, x2: default_rewrite(x1, x2), new_args, args),
+        match y:
+            case LogicExpression() as expr:
```

Comment:
Again, let's avoid making this change. Rewriters are intended to work on more than just LogicExpr. let's leave this as it was.